client/state/themes/utils.js

```diff
@@ -204,31 +204,34 @@ export function isWpcomTheme( themeId ) {
 
 /**
  * Returns a filtered themes array. Filtering is done based on particular themes
- * matching provided query and isWpocmTheme predicate.
+ * matching provided query and isWpocmTheme predicate if filterWpcom is true
  * Themes on Jetpack installed from WordPress.com have -wpcom suffix
- * We filter out all those themes because they will also be visible on
- * second list specific to WordPress.com. This may be too simple aproach
- * so it may revisit it again later.
+ * We can filter out all those themes because they will also be visible on
+ * second list specific to WordPress.com for Jetpack versions from 4.4.2.
+ * This may be too simple aproach so it may revisit it again later.
  *
  * TODO Veriy that wpcom filtering is sufficien.
  *
- * @param  {Array}  themes Array of theme objects
- * @param  {Object} query  Themes query
- * @return {Array}         Filtered themes
+ * @param  {Array}   themes      Array of theme objects
+ * @param  {Object}  query       Themes query
+ * @param  {Boolean} filterWpcom Switch to filter out themes with -wpcom suffix
+ * @return {Array}               Filtered themes
  */
-export function filterThemesForJetpack( themes, query ) {
+export function filterThemesForJetpack( themes, query, filterWpcom = false ) {
```

Comment:
Hrm. I'm not too happy about putting two different kinds of filter into the same function and switch one of them on and off with a flag. I'd rather separate concerns, i.e. have a separate filter function for each, and use them subsequently.